utils/biospecimen.js

```diff
@@ -723,10 +829,15 @@ const biospecimenAPIs = async (req, res) => {
         }
         let requestData = req.body;
         if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
-        const { assignKitToParticipants } = require('./firestore');
-        const response = await assignKitToParticipants(requestData);
-        if(!response) return res.status(404).json(getResponseJSON('ERROR!', 404));
-        return res.status(200).json({message: `Success!`, code:200})
+        const { assignKitToParticipant } = require('./firestore');
+        const response = await assignKitToParticipant(requestData);
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+           return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({data: response, code:200});
+        }
```

Comment:
Please consider the same as above for structuring the error handling flow in the endpoint and the function it calls.